homeassistant/components/bluetooth_le_tracker/device_tracker.py

```diff
@@ -26,6 +27,13 @@ def setup_scanner(hass, config, see, discovery_info=None):
     # pylint: disable=import-error
     import pygatt
     new_devices = {}
+    adapter = None
+
+    async def async_stop():
+        if adapter is not None:
```

Comment:
I think you need `nonlocal adapter` somewhere for this to ever become true(?)

Reply:
absolutely right, what a silly mistake